src/manual_wrappers.jl

```diff
@@ -30,13 +30,7 @@ end
 
 
 ## Root finding
-function gsl_function_helper(x::Cdouble, fn)::Cdouble
-    try
-        return fn(x)
-    catch
-        return NaN
-    end
-end
+gsl_function_helper(x::Cdouble, fn::F) where {F} = fn(x)
```

Comment:
This means that the user / callback function is responsible for making sure there's no error being thrown. Personally this is my perferred tradeoff...

Also I don't think you need the `::F` and the `::Cdouble` is still helpful and shouldn't hurt performance.